pkg/storage/sqlcommon/sqlcommon.go

```diff
@@ -17,10 +17,14 @@ import (
 	"github.com/openfga/openfga/pkg/logger"
 	"github.com/openfga/openfga/pkg/storage"
 	tupleUtils "github.com/openfga/openfga/pkg/tuple"
+	"github.com/pressly/goose/v3"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+// TODO everytime we add a migration we have to update this. Is there a better way?
+const latestDBVersion = 5
```

Comment:
@miparnisari maybe we should get these by looking at `assets.EmbedMigrations` for the configured datastore? That would prevent us from forgetting to bump this number while also still working if the number diverges between datastores.

Reply:
I'd put this along with the build info in `build/build.go`. Each OpenFGA build expects a particular database revision, and a database revision is really a range, not a singular revision. Multiple versions of a build can run on different revisions of a database in a compatible way. So I'd recommend we have `build.MinRequiredDatabaseRevision` and `build.MaxRequiredDatabaseRevision`.

On startup if we see that the database revision is < Min or > Max then we fail, the ready check can also make sure we stay within the range. If we are within the range but not at the latest "HEAD" revision, then we should consider emitting a warning, because their may be indexes or other things that you are missing if you are not running the maximum revision supported for that build.